devices/ICub/src/ICub.cpp

```diff
@@ -0,0 +1,345 @@
+        , icubImpl(impl)
+    {
+        // Set the sensor status Ok after receiving the first data on the wrench ports
+        if (!icubImpl->leftHandFTPort.firstRun || !icubImpl->rightHandFTPort.firstRun) {
+            auto nonConstThis = const_cast<ICubForceTorque6DSensor*>(this);
+            nonConstThis->setStatus(wearable::sensor::SensorStatus::Ok);
+        }
+    }
+
+    ~ICubForceTorque6DSensor() override = default;
+
+    void setStatus(const wearable::sensor::SensorStatus status) { m_status = status; }
+
+    // ==============================
+    // IForceTorque6DSensor interface
+    // ==============================
+    bool getForceTorque6D(Vector3& force3D, Vector3& torque3D) const override
+    {
+        assert(icubImpl != nullptr);
+
+        std::vector<double> wrench;
+
+        // Reading wrench from WBD ports
+        if (this->m_name
+            == icubImpl->wearableName + wearable::Separator
+                   + sensor::IForceTorque6DSensor::getPrefix() + "leftWBDFTSensor") {
+
+            std::lock_guard<std::mutex> lck(icubImpl->leftHandFTPort.mtx);
+            wrench = icubImpl->leftHandFTPort.wrenchValues;
+
+            icubImpl->timeStamp.time = yarp::os::Time::now();
+        }
+        else if (this->m_name
+                 == icubImpl->wearableName + wearable::Separator
+                        + sensor::IForceTorque6DSensor::getPrefix() + "rightWBDFTSensor") {
+
+            std::lock_guard<std::mutex> lck(icubImpl->rightHandFTPort.mtx);
+            wrench = icubImpl->rightHandFTPort.wrenchValues;
+
+            icubImpl->timeStamp.time = yarp::os::Time::now();
+        }
+
+        // Check the size of wrench values
+        if (wrench.size() == 6) {
+            force3D[0] = wrench.at(0);
+            force3D[1] = wrench.at(1);
+            force3D[2] = wrench.at(2);
+
+            torque3D[0] = wrench.at(3);
+            torque3D[1] = wrench.at(4);
+            torque3D[2] = wrench.at(5);
+        }
+        else {
+            yWarning() << LogPrefix << "Size of wrench read is wrong. Setting zero values instead.";
+            force3D.fill(0.0);
+            torque3D.fill(0.0);
+
+            // Set sensor status to error if wrong data is read
+            // auto nonConstThis = const_cast<ICubForceTorque6DSensor*>(this);
+            // nonConstThis->setStatus(wearable::sensor::SensorStatus::Error);
+        }
+
+        return true;
+    }
+};
+
+// ==========================================
+// ICub constructor/destructor implementation
+// ==========================================
+ICub::ICub()
+    : pImpl{new ICubImpl()}
+{}
+
+ICub::~ICub() = default;
+
+// ======================
+// DeviceDriver interface
+// ======================
+bool ICub::open(yarp::os::Searchable& config)
+{
+    yInfo() << LogPrefix << "Starting to configure";
+
+    // Configure clock
+    if (!yarp::os::Time::isSystemClock()) {
+        yarp::os::Time::useSystemClock();
+    }
```

Comment:
Yes, It seems fishy also to me. Why is that used?

Reply:
I agree on the usage of network clock in case of using icub in gazebo! Ideally, at this stage, this device should connect to the wholebodydynamics ports at the hands end-effectors and pass the FT values as a wearable device. So, we can remove the explicit configuration of this device to the system clock.

Reply:
@Yeshasvitvs Has this been fixed?

Reply:
done in https://github.com/robotology/wearables/pull/25/commits/42eba1b291ebfda8a27629c4a999cf1225bdca30

Reply:
done in https://github.com/robotology/wearables/pull/25/commits/42eba1b291ebfda8a27629c4a999cf1225bdca30